banano-quest/banano-quest/ViewControllers/CompleteQuestViewController.swift

```
@@ -10,41 +10,103 @@ import UIKit
+        mapView.showsUserLocation = true
+        mapView.showsCompass = true
+        
+        // Location Manager settings
+        locationManager.delegate = self
+        locationManager.desiredAccuracy = kCLLocationAccuracyNearestTenMeters
+        locationManager.requestWhenInUseAuthorization()
+        locationManager.requestAlwaysAuthorization()
+        
     }
     
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
-        
+        // Background settings
         bananoBackground.layer.cornerRadius = bananoBackground.frame.size.width / 2
         bananoBackground.clipsToBounds = true
+        
+        // Checks is location services are enabled to start updating location
+        if CLLocationManager.locationServicesEnabled() {
+            locationManager.startUpdatingLocation()
+        }else{
+            print("Location services are disabled, please enable before trying again.")
+        }
+        
+        // Refresh view
+        refreshView()
+    }
+    
+    func refreshView() {
+        // Details view
+        bananoCountLabel.text = "0/\(quest?.maxWinners ?? 1)"
+        prizeValueLabel.text = "\(quest?.prize ?? 1).0 USD"
+        // TODO: Get location from merkleRoot
+        distanceValueLabel.text = "20M"
+        questDetailTextView.text = quest?.hint
     }
     
     override func didReceiveMemoryWarning() {
         super.didReceiveMemoryWarning()
         // Dispose of any resources that can be recreated.
     }
     
-
-    /*
-    // MARK: - Navigation
-
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destinationViewController.
-        // Pass the selected object to the new view controller.
+    func locationManager(_ manager: CLLocationManager, didFailWithError error: Error) {
+        print(error)
```

Show error to the user?

---

This should be a silent error